tools/testsys/src/aws_resources.rs

```diff
@@ -56,18 +59,56 @@ impl AwsK8s {
                 self.kube_arch(),
                 self.kube_variant(),
             )?;
+            // check for existing cluster crd
+            let cluster_exists = client
+                .resource_client()
+                .get(&cluster_name)
+                .await
+                .allow_not_found(|_| ())?
+                .is_some();
+            if !cluster_exists {
+                debug!("Cluster crd does not exist");
+                crds.push(self.eks_crd(cluster_name, testsys_images, &test)?)
+            }
+            // check for conflicting resources (ones that use the same cluster)
+            let conflicting_resources: Vec<String> = if cluster_exists {
```

Comment:
I saw that `conflicting_resources` is created here, but then passed down to `sonobuoy_test_crds` -> `ec2_crd` -> `Resource { conflicts_with }`. There could be a way to increase cohesion and pass down the reference only once, but it will require some refactoring, obviously not in this PR.

For this concern, the rusteceans in the team could weight in, is passing down dependencies like this until the method where they are used OK in rust?

Reply:
The best I could come up with was passing `&[String]` instead of `Vec<String>`, maybe others have a better idea?

Reply:
My instinct here would be to create a new type, like the `CrdInput` type I mentioned, and make `conflicting_resources` a method on that type, assuming it has all the fields it needs to handle the query.

That way, you can call the method where it's ultimately used, rather than at the outset where you have to pass it down into the call chain.